### PR TITLE
feat/244 query param sanitization

### LINKED ISSUE #244: Insufficient Sanitization of Parameters
Relates to: https://github.com/anoma/namada-interface/issues/222

Make sure we are doing proper params sanitization in the extension. 

Check out `query.get` calls.
Check out query params in the interface routes.

### PR DESCRIPTION
**Pull request by emccorson**
This PR removes unused code flagged as problematic in the audit, and wraps other potentially unsafe functions in the `sanitize` function from [dompurify](https://github.com/cure53/DOMPurify) which strips strings of anything that could be potentially used in an XSS attack.

- feat: sanitize URL params in interface
- feat: sanitize location in interface             
- feat: sanitize query params in extension
- feat: allow URL navigation in extension by adding HashRouter
- refactor: remove unused getUrl and associated functions

Fixes #244.

### CODE REVIEW

**Review by mateuszjasiuk — APPROVED** [review]
LGTM :)

**Comment by emccorson** [comment]
Holding off merge until #266 is merged so I can move the hooks from this PR to the new hooks package.

**Review by jurevans — APPROVED** [review]
Nice! Looks good

**Pull request merged**
